src/groundcrew/tools.py

```diff
@@ -350,3 +370,298 @@ def __call__(self, user_prompt: str, include_code: bool) -> str:
         prompt += f'{user_prompt}\n\n'
 
         return self.llm(prompt)
+
+
+class CyclomaticComplexityTool:
+    """
+    Tool for computing the cyclomatic complexity of a codebase.
+    """
+    def __init__(
+            self,
+            base_prompt: str,
+            collection: Collection,
+            llm: Callable,
+            min_max_complexity: int = 11
+        ):
+        self.collection = collection
+        self.llm = llm
+        self.base_prompt = base_prompt
+        self.min_max_complexity = min_max_complexity
+
+    def __call__(
+            self,
+            user_prompt: str,
+            filepath_inexact: str = 'none',
+            sort_on: str = 'max'
+        ) -> str:
+        """Answer questions using about the complex parts of a codebase.
+        This method will create a summary of the most complex files to help answer
+        the question.
+
+        Args:
+            user_prompt (str): The user's question.
+            filepath_inexact (str): A filepath explicitly requested by the user.  If the
+              user did not explicitly request a filepath, pass 'none'.
+            sort_on (str): Sort the results by the "average" or "max" complexity of the file.
+        """
+
+        if filepath_inexact != 'none':
+            filepath = fuzzy_match_file_path(self.collection, filepath_inexact, 50)
+
+            if filepath is None:
+                return f'Could not find a source file matching `{filepath_inexact}`'
+
+            all_items = self.collection.get(
+                include=['metadatas'],
+                where={'filepath': filepath}
+            )
+
+            files = {
+                id_: metadata['text']
+                for id_, metadata in zip(all_items['ids'], all_items['metadatas'])
+                if id_[-3:] == '.py'
+            }
+
+        else:
+            files = get_python_files(self.collection)
+
+        sort_on = sort_on.lower()
```

Comment:
Suggest changing this to `sort_on = str(sort_on).lower()`

The LLM passed `None` (not sure how?), which resulted in the error:

```
How complex is openaiapi.py?
⠋ Running CyclomaticComplexityToolPlease standby while I run the tool CyclomaticComplexityTool...
("Your task is to analyze the cyclomatic complexity summary of 'openaiapi.py' and provide the complexity information of the file.", {'filepath_inexact': 'openaiapi.py', 'sort_on': None})

sort_on = sort_on.lower()
              ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'lower'
```

Reply:
@cameronfabbri I believe `None`s are getting in from `Agent.run_tool` which fills in missing parameters with `None`

Reply:
Oh yeah, I wrote that piece of code 😆